### PR TITLE
traffic_ctl - plugin msg  now require only the tag as mandatory field data field is now optional.

### PR DESCRIPTION
**Pull request by brbzull0**
`traffic_ctl` can send a plugin message, which has a tag and a body. The CLI requires a tag and a body, but in some cases the body is irrelevant. However the CLI will fail if there’s no body. This change make the body/data optional, if not passed then the message handler will get data size as 0 and the data as NULL.

### CODE REVIEW

File: example/plugins/c-api/lifecycle_plugin/lifecycle_plugin.c [hunk 1]
```diff
@@ -49,6 +49,10 @@ CallbackHandler(TSCont this, TSEvent id, void *data)
   case TS_EVENT_LIFECYCLE_MSG: {
     TSPluginMsg *msg = (TSPluginMsg *)data;
     TSDebug(PLUGIN_NAME, "Message to '%s' - %zu bytes of data", msg->tag, msg->data_size);
+    if (msg->data_size == 0) {
```
Comment (SolidWallOfCode) [inline example/plugins/c-api/lifecycle_plugin/lifecycle_plugin.c:52]: Isn't this clear from the message on line 51?

File: plugins/experimental/memory_profile/memory_profile.cc [hunk 2]
```diff
@@ -49,26 +49,28 @@ CallbackHandler(TSCont cont, TSEvent id, void *data)
     TSDebug(PLUGIN_NAME, "Message to '%s' - %zu bytes of data", msg->tag, msg->data_size);
     if (strcmp(PLUGIN_NAME, msg->tag) == 0) { // Message is for us
 #if TS_HAS_JEMALLOC
-      int retval = 0;
-      if (strncmp((char *)msg->data, "dump", msg->data_size) == 0) {
-        if ((retval = mallctl("prof.dump", nullptr, nullptr, nullptr, 0)) != 0) {
-          TSError("mallct(prof.dump) failed retval=%d errno=%d", retval, errno);
+      if (msg->data_size) {
```
Comment (SolidWallOfCode) [inline plugins/experimental/memory_profile/memory_profile.cc:52]: Is the size check needed here, or just in the debug message at line 72? `strncmp` should handle a zero size.

Comment (brbzull0) [reply plugins/experimental/memory_profile/memory_profile.cc:52]: strncmp:
>  The behavior is undefined when either lhs or rhs is the null pointer.

Not sure if the size = 0 will stop the function to check the ptr. Probably is safe this way.

File: example/plugins/c-api/lifecycle_plugin/lifecycle_plugin.c (hunk 1 above)
Comment (brbzull0) [reply example/plugins/c-api/lifecycle_plugin/lifecycle_plugin.c:52]: yes, just want to make a point as this is an example plugin.

**Review by randall — APPROVED** [review]
looks reasonable to me.

**Pull request merged**